x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -337,40 +338,49 @@ func (m *MetricSet) reportEvent(ms mapstr.M, reporter mb.ReporterV2, qry string)
 	return ok
 }
 
-// inferTypeFromMetrics to organize the output event into 'numeric', 'strings', 'floats' and 'boolean' values
-// so we can dynamically map all fields inside those categories
 func inferTypeFromMetrics(ms mapstr.M) mapstr.M {
 	ret := mapstr.M{}
 
 	numericMetrics := mapstr.M{}
 	stringMetrics := mapstr.M{}
 	boolMetrics := mapstr.M{}
+	dateMetrics := mapstr.M{}
+	objectMetrics := mapstr.M{}
 
 	for k, v := range ms {
-		switch v.(type) {
-		case float64:
-			numericMetrics[k] = v
+		switch val := v.(type) {
+		case float32, float64, int16, int32, int64, uint32, uint64:
```

Comment:
Add all those types present in [numeric](https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html)